_episodes/06-raster-intro.md

```diff
@@ -0,0 +1,462 @@
+- "It is good to consider using nan to represent no data value(s), for statistics and visualization. "
+- "The `DataArray.plot()` function can be used to visualize single-band raster. Consider using the `robust=True` option for a better color limit."
+- "The `DataArray.plot.imshow()` function can be used to visualize multi-band raster."
+---
+
+> ## Things You'll Need To Complete This Episode
+>
+> See the [lesson homepage]({{ site.baseurl }}) for detailed information about the software,
+> data, and other prerequisites you will need to work through the examples in this episode.
+{: .prereq}
+
+In this episode, we will introduce the fundamental principles, packages and
+metadata/raster attributes that are needed to work with raster data in Python. We will discuss some of the core metadata elements that we need to understand to work with rasters, including Coordinate Reference Systems, no data values, and resolution. We will also explore missing and bad data values as stored in a raster and how Python handles these elements.
+
+We will use a Python package in this episode to work with raster data -
+[`rioxarray`](https://corteva.github.io/rioxarray/stable/),
+which is based on the popular [`rasterio`](https://rasterio.readthedocs.io/en/latest/) package for working with rasters and [`xarray`](http://xarray.pydata.org/en/stable/) for working with multi-dimensional arrays.
+
+We will also use the [`pystac`](https://github.com/stac-utils/pystac) package to load raster from the searching results from the exercise of the previous episode. This would not be necessary if you skipped it.
+
+Make sure that you have `rioxarray` and `pystac` (optional) installed and imported.
+
+~~~
+import rioxarray
+import pystac # Not required if you skipped the previous episode 
+~~~
+{: .language-python}
+
+
+> ## Introduce the Data
+>
+> In this episode, we will continue from the `search.json` file, which we saved in an exercise from the previous episode. We will use one scene in the searching results as an example, and demonstrate the data loading and visualization.
+>
+> If you have skipped the previous episode on data accessing, you can download the same raster data from the [data repository](ToDo: add link) of this lesson.
+{: .callout}
+
+## Load a Raster and View Attributes
+
+In the previous episode, we did an exercise searching Sentinel-2 images, and then saved the searching results to `search.json`. This file contains the information on where and how to access the target images from a remote repository. We can use the function `pystac.ItemCollection.from_file()` to load the searching results as an `Item` list.
+
+
+~~~
+items = pystac.ItemCollection.from_file("search.json")
+[item for item in items]
+~~~
+{: .language-python}
+~~~
+[<Item id=S2A_31UFU_20200328_0_L2A>,
+ <Item id=S2B_31UFU_20200326_0_L2A>,
+ <Item id=S2B_31UFU_20200323_0_L2A>,
+ <Item id=S2A_31UFU_20200321_0_L2A>]
```

Comment:
```suggestion
[<Item id=S2A_31UFU_20200328_0_L2A>, <Item id=S2B_31UFU_20200326_0_L2A>]
```

Reply:
I agree with Francesco on keeping the details to the minimum. Hence I will remove the details of the Items.